app/model/User.scala

```diff
@@ -1,33 +1,61 @@
 package model
 
 import play.api.libs.json.{Json, Writes}
+import com.okta.sdk.resource.model.User as OktaUser
 
 case class User(
-    identityId: String,
-    brazeId: Option[String],
+    oktaId: String,
+    legacyIdentityId: String,
     emailAddress: Option[String],
     emailValidated: Option[Boolean],
     userName: Option[String],
     name: Name,
     address: Address,
     phoneNumber: Option[PhoneNumber],
     registrationLocation: Option[String],
-    permissions: Seq[Permission],
+    permissions: Seq[Permission]
 )
 
 object User {
+
+  def fromOktaUser(oktaUser: OktaUser): User = {
+    User(
+      oktaId = oktaUser.getId,
+      legacyIdentityId = requiredCustomProfileField(oktaUser, "legacyIdentityId"),
+      emailAddress = optionalProfileField(oktaUser.getProfile.getEmail),
+      emailValidated = customProfileField(oktaUser, "emailValidated").flatMap(_.toBooleanOption),
+      userName = None,
+      name = Name.fromOktaUser(oktaUser),
+      address = Address.fromOktaUser(oktaUser),
+      phoneNumber = PhoneNumber.fromOktaUser(oktaUser),
+      registrationLocation = customProfileField(oktaUser, "registrationLocation"),
+      permissions = Nil
+    )
+  }
+
+  def optionalProfileField(fieldValue: String): Option[String] =
+    Option(fieldValue).filter(!_.isBlank)
```

Comment:
tiny: maybe this could be called toOption or NoneIfEmpty or something like that. It doesn't really operate on profile fields like the others, it just transforms a string

Reply:
That's true

Reply:
Done in 6b5d5be55ae67cd8c7c407c21a421f829573de75